pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_class_controller.go

```diff
@@ -0,0 +1,115 @@
+package gatewayapi
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	kube_apierrs "k8s.io/apimachinery/pkg/api/errors"
+	kube_types "k8s.io/apimachinery/pkg/types"
+	kube_ctrl "sigs.k8s.io/controller-runtime"
+	kube_client "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	kube_handler "sigs.k8s.io/controller-runtime/pkg/handler"
+	kube_reconcile "sigs.k8s.io/controller-runtime/pkg/reconcile"
+	kube_source "sigs.k8s.io/controller-runtime/pkg/source"
+	gatewayapi "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+// GatewayClassReconciler reconciles a GatewayAPI GatewayClass object.
+type GatewayClassReconciler struct {
+	kube_client.Client
+	Log logr.Logger
+}
+
+const finalizerName = "gateways.kuma.io/finalizer"
+
+const gatewayClassKey = "metadata.gatewayClass"
```

Comment:
I don't think it matters, but other indexing keys have a leading `.`; suggest we make this consistent so that future readers don't think it is special somehow.